src/services/BackgroundValidatePeers.ts

```diff
@@ -0,0 +1,46 @@
+import { Injector } from 'reduct'
+import PeerDatabase from './PeerDatabase'
+import Config from './Config'
+import axios from 'axios'
+import { create as createLogger } from '../common/log'
+import { shallowValidatePeer } from '../util/validatePeer'
+const log = createLogger('BackgroundValidatePeers')
+const CronJob = require('cron').CronJob
+
+export default class BackgroundValidatePeers {
+  private peerDb: PeerDatabase
+  private config: Config
+  constructor (deps: Injector) {
+    this.peerDb = deps(PeerDatabase)
+    this.config = deps(Config)
+  }
+
+  start () {
+    const db = this.peerDb
+    const config = this.config
+    const job = new CronJob({
+      cronTime: '00 30 2 * * *',
+      onTick: function () {
+        /*
+        * Runs every day at 2:30:00 AM.
```

Comment:
We shouldn't be running this at a well defined time. This would lead to a thundering herd problem and the network could essentially DDoS itself every night. Instead this should just run on a timeout of a couple of hours but with the specific amount of time slightly randomized to stagger the timing.